### PR TITLE
Fix for Issue EZP-25301

### PR DESCRIPTION
**Pull request by pkamps**
Link: https://jira.ez.no/browse/EZP-25301

With eZContentUpload you can use custom UploadHandler. eZContentUpload will then call the method 'handleFile' of your custom UploadHandler. One of the method parameters is the original file name "originalFilename". eZContentUpload is not sending the correct value for this parameter -- this PR is fixing it.

### CODE REVIEW

File: kernel/classes/ezcontentupload.php [hunk 1]
```diff
@@ -246,7 +246,7 @@ function handleLocalFile( &$result, $filePath, $location, $existingNode, $nameSt
         // The handler will be responsible for the rest of the execution.
         if ( is_object( $handler ) )
         {
-            $originalFilename = $filePath;
+            $originalFilename = $nameString ? $nameString : $filePath;
```
Comment (crevillo) [inline kernel/classes/ezcontentupload.php]: shouldn't we check if $nameString is ```!= ''```?

**Comment by yannickroger** [comment]
Hi,
If you want to get this merged, you should:
-  add some steps to reproduce in your jira issue.
- fill up the description of the PR with what it is about and how you tested it.
- squash your commits.

File: kernel/classes/ezcontentupload.php (hunk 1 above)
Comment (pkamps) [inline kernel/classes/ezcontentupload.php]: yes, that's true

**Comment by pkamps** [comment]
Steps added to the jira issue.
Description to the PR added.

Could you explain why how and why I would need to squash my commits, thanks?

**Comment by yannickroger** [comment]
Regarding why squash commit, it is to keep a clean and readable git history.
So, your fix should be in a commit called something like: `Fix EZP-25031: eZContentUpload is not providing original file name to custom UploadHandler`

How to clean it up, I can help you here or you can ping me on slack. (If you prefer the RTFM way :) what to look for is `git rebase -i` and `push --force`)

**Comment by pkamps** [comment]
Closed in favor of https://github.com/ezsystems/ezpublish-legacy/pull/1228 which only contains a single commit.

Thanks for explaining the squashing. The changes are so simple that I decide to create a new pull request.

**Pull request closed without merging**